pkg/frameworks/custom/custom_container.go

```diff
@@ -1,10 +1,26 @@
 package custom
 
 import (
+	"fmt"
+
+	knserving "github.com/knative/serving/pkg/apis/serving"
 	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func CreateCustomContainer(customSpec *v1alpha1.CustomSpec) *v1.Container {
-	return &customSpec.Container
+type CustomFramework struct {
```

Comment:
Do we need another layer on top ? I think we can implement the interface directly on spec

Reply:
Yes this made me sad too... I did this because it was moaning and complaining about non-local types... however on further research it turns out all is well if I use a type alias... so yay!